sway-ir/src/optimize/simplify_cfg.rs

```diff
@@ -80,25 +115,20 @@ fn merge_blocks(context: &mut Context, function: &Function) -> Result<bool, IrEr
     // There may be more blocks which are also singly paired with these twins, so iteratively
     // search for more blocks in a chain which can be all merged into one.
     loop {
-        match block_terms_with_br(block_chain.last().copied().unwrap()) {
+        match check_candidate(block_chain.last().copied().unwrap()) {
             None => {
                 // There is no twin for this block.
                 break;
             }
             Some(next_pair) => {
-                if are_uniquely_paired(&next_pair) {
-                    // Add the next `to_block` to the chain and continue.
-                    block_chain.push(next_pair.1);
-                } else {
-                    // The chain has ended.
-                    break;
-                }
+                block_chain.push(next_pair.1);
             }
         }
     }
 
     // Keep a copy of the final block in the chain so we can adjust the successors below.
     let final_to_block = block_chain.last().copied().unwrap();
+    let final_to_block_succs = final_to_block.successors(context);
```

Comment:
Does this need to be up here, or can we declare it down below where it's used?

Reply:
> Does this need to be up here, or can we declare it down below where it's used?

This needs to be here because, after merging, the terminator is moved to a different block.